frontend/src/components/MeetingRoom/ToolbarOverflowMenu/ToolbarOverflowMenu.tsx

```diff
@@ -92,19 +100,22 @@ const ToolbarOverflowMenu = ({
                   setIsEmojiGridOpen={setIsEmojiGridOpen}
                   isParentOpen={isToolbarOverflowMenuOpen}
                 />
-                <ArchivingButton />
+                <ArchivingButton handleCloseMenu={handleCloseMenu} />
                 {isReportIssueEnabled && (
                   <ReportIssueButton
                     isOpen={rightPanelActiveTab === 'issues'}
-                    handleClick={toggleReportIssue}
+                    handleClick={handleCloseMenu(toggleReportIssue)}
```

Comment:
We need to pass these in as functions, not immediately invoking them. I believe these are being called immediately on rendering, which probably does nothing right now, but could cause some strangeness down the line.

Reply:
No I think this is right, `handleCloseMenu` is returning a function, so this will not execute the inner function on rendering. Though the Archiving button prop wrapper is not being invoked in the props so that one won't work unless it's being invoked twice in the child component